mathesar/api/exceptions/exceptions.py

```diff
@@ -44,3 +56,331 @@ class GenericAPIError(MathesarAPIException):
     def __init__(self, error_body_list, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR):
         self.detail = [error_body._asdict() for error_body in error_body_list]
         self.status_code = status_code
+
+
+class UniqueViolationAPIException(MathesarAPIException):
+    def __init__(
+            self,
+            exception,
+            error_code=ErrorCodes.UniqueViolation.value,
+            message="This column has non-unique values so a unique constraint cannot be set",
+            field=None,
+            details=None,
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
+    ):
+        exception_detail = get_default_exception_detail(exception, error_code, message, field, details)._asdict()
+        self.detail = [exception_detail]
+        self.status_code = status_code
+
+
+class NotFoundAPIException(MathesarAPIException):
+    def __init__(
+            self,
+            exception,
+            error_code=ErrorCodes.NotFound.value,
+            message=None,
+            field=None,
+            details=None,
+            status_code=status.HTTP_404_NOT_FOUND
```

Comment:
Moved the status code out of `init`. Some Exceptions have `status_code` in their `init` so that it can be overridden. I refactored and moved the exceptions into proper modules, so that consistency can be maintained.